### PR TITLE
Add mocked enrollment UI for the 2FA App Authenticator method

### PR DESCRIPTION
**Pull request by dblandin**
Introduce the 2FA App Authenticator enrollment UI. Hooks up the modal and form components including basic validation.

The OTP code `123456` will mimic a successful enrollment. Anything else will mimic an invalid OTP code.

Screenshot from 2020-04-14 20-21-38

### CODE REVIEW

File: src/Components/UserSettingsTwoFactorAuthentication/Components/AppSecondFactorMethod.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { BorderBox, Button, Flex, Link, Sans, Serif } from "@artsy/palette"
+import { BorderBoxProps } from "@artsy/palette/dist/elements/BorderBox/BorderBoxBase"
```
Comment (zephraph) [inline src/Components/UserSettingsTwoFactorAuthentication/Components/AppSecondFactorMethod.tsx:2]:
```suggestion
import { BorderBox, BorderBoxProps, Button, Flex, Link, Sans, Serif } from "@artsy/palette"
```

Comment (dblandin) [reply src/Components/UserSettingsTwoFactorAuthentication/Components/AppSecondFactorMethod.tsx:2]: I was also surprised by this error message:

```
import BorderBoxProps

Module '"../../../../node_modules/@artsy/palette/dist"' has no exported member 'BorderBoxProps'.ts(2305)
```

File: src/Components/UserSettingsTwoFactorAuthentication/Components/AppSecondFactorMethod.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    if (values.code === "123456") {
+      setShowSetupModal(false)
+    } else {
+      actions.setSubmitting(false)
+      actions.setFieldError("code", "Invalid two-factor authentication code.")
+    }
+  }
+
+  return (
+    <BorderBox p={20} {...props}>
```
Comment (zephraph) [inline src/Components/UserSettingsTwoFactorAuthentication/Components/AppSecondFactorMethod.tsx:30]: Does `p={2}` work here?

Comment (zephraph) [reply src/Components/UserSettingsTwoFactorAuthentication/Components/AppSecondFactorMethod.tsx:30]:
```suggestion
    <BorderBox p={2} {...props}>
```

File: src/Components/UserSettingsTwoFactorAuthentication/Components/AppSecondFactorMethod.tsx (hunk 1 above)
Comment (zephraph) [reply src/Components/UserSettingsTwoFactorAuthentication/Components/AppSecondFactorMethod.tsx:2]: Yep, you're absolutely right. See https://github.com/artsy/palette/blob/master/packages/palette/src/elements/BorderBox/BorderBox.tsx. 

We'll  fix it later. 👍

**Comment by artsyit** [comment]
:rocket: PR was released in `v26.15.0` :rocket:

**Pull request merged**